src/rename-folder.ts

```
@@ -0,0 +1,46 @@
+import { util } from "lspconfig";
+import { renameFile } from "@ts/rename-file";
+import { debugLog } from "@ts/utils";
+
+interface Opts {
+  force?: boolean;
+}
+
+export const renameFolder = (
+  source: string,
+  target: string,
+  opts: Opts = {}
+): boolean => {
+  debugLog(source, target);
+  const sourceBufnr = vim.fn.bufadd(source);
+  vim.fn.bufload(sourceBufnr);
+
+  if (!util.path.is_dir(source)) {
+    debugLog("source is not a directory");
+    return false;
+  }
+
+  if (
+    util.path.exists(target) &&
+    util.path.is_dir(target) &&
```

I think this check may cause issues, since we are not handling the case where the target exists but is not a directory (which should just throw an error).